wgsl/index.bs

```diff
@@ -90,7 +90,89 @@ that run on the GPU.
  * Features and semantics are exactly the ones of SPIR-V
  * Each item in this spec *must* provide the mapping to SPIR-V for the construct
 
-## Technical Overview TODO ## {#technical-overview}
+## Technical Overview ## {#technical-overview}
+
+WebGPU issues a unit of work to the GPU in the form of a <dfn noexport>provoking command</dfn>.
+[SHORTNAME] is concerned with two kinds of provoking commands:
+* a <dfn noexport>draw command</dfn> executes a [=GPURenderPipeline|render pipeline=]
+    in the context of [=pipeline input|inputs=], [=pipeline output|outputs=], and attached [=resources=].
+* a <dfn noexport>dispatch command</dfn> executes a [=GPUComputePipeline|compute pipeline=]
+    in the context of [=pipeline input|inputs=] and attached [=resources=].
+
+Both kinds of pipelines use shaders written in [SHORTNAME].
+
+A <dfn noexport>shader</dfn> is the portion of a [SHORTNAME] program that executes a [=shader stage=] in a pipeline.
+A shader comprises:
+* An [=entry point=] [=function/function=].
+* The transitive closure of all called functions, starting with the entry point.
+    This set includes both [=user-defined function|user-defined=] and [=built-in function|built-in=] functions.
+    (For a more rigorous definition, see "[=functions in a shader stage=]".)
+* The set of variables and constants [=statically accessed=] by all those functions.
+* The set of types used to define or analyze all those functions, variables, and constants.
+
+When executing a shader stage, the implementation:
+* Binds [=resources=] to variables in the shader's [=resource interface of a shader|resource interface=],
```

Comment:
What about function and private variables?

Reply:
Right.  We definitely need to address the non-resource module-scope variables: private, workgroup.
The function ones are a judgement call.  It's not clear the implementation must allocate storage for those before invocation; but must ensure that each variable's storage is distinct during its lifetime.
I'll update with something.

Reply:
I think for the function-storage variables, the activity of allocating them is seen as a responsibility of the execution of the function itself. Sure the allocations could all be done statically, but that's an implementation choice.

Reply:
PTAL